File: src/main/java/org/social/social_network/entity/Comment.java
package org.social.social_network.entity;

import lombok.Data;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.relational.core.mapping.Table;

import java.time.Instant;
import java.util.UUID;

@Data
@Table(name = "comments")
public class Comment {

    @Id
    private UUID id;

    private String text;

    @CreatedDate
    private Instant pubDate = Instant.now();

    @LastModifiedDate
    private Instant modifiedDate = Instant.now();

    private UUID authorId;

    private UUID postId;
}

File: src/main/java/org/social/social_network/entity/Comment.kt
package org.social.social_network.entity

import lombok.Data
import org.springframework.data.annotation.CreatedDate
import org.springframework.data.annotation.Id
import org.springframework.data.annotation.LastModifiedDate
import org.springframework.data.relational.core.mapping.Table
import java.time.Instant
import java.util.*

@Data
@Table(name = "comments")
class Comment {
    @Id
     val id: UUID? = null

     val text: String? = null

    @CreatedDate
     val pubDate: Instant = Instant.now()

    @LastModifiedDate
     val modifiedDate: Instant = Instant.now()

    val authorId: UUID? = null

     val postId: UUID? = null
}